File: Python/Swap Nodes in Pairs .py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param a ListNode
    # @return a ListNode
    def swapPairs(self, head):
        if not head:
            return None
        if not head.next:
            return head
        
        dummy = ListNode(0)
        dummy.next = head
        cur = dummy
        
        while cur.next and cur.next.next:
            cur.next = self.swap(cur.next, cur.next.next)
            cur = cur.next.next
                
        return dummy.next   
        
    def swap(self, node1, node2):
        node1.next = node2.next
        node2.next = node1
        return node2

        

File: Java/src/Swap Nodes in Pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) return head;
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode ret = dummy;
        while(ret.next != null && ret.next.next != null){
            ret.next = swap(ret.next, ret.next.next);
            ret = ret.next.next;
        }
        return dummy.next;
    }
        
    public ListNode swap(ListNode node1, ListNode node2){
        node1.next = node2.next;
        node2.next = node1;
        return node2;
    }
}